src/main/java/org/broadinstitute/consent/http/service/dao/DarCollectionServiceDAO.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.consent.http.service.dao;
+
+import com.google.inject.Inject;
+import org.broadinstitute.consent.http.db.UserDAO;
+import org.broadinstitute.consent.http.enumeration.ElectionStatus;
+import org.broadinstitute.consent.http.enumeration.ElectionType;
+import org.broadinstitute.consent.http.enumeration.UserRoles;
+import org.broadinstitute.consent.http.enumeration.VoteType;
+import org.broadinstitute.consent.http.models.DarCollection;
+import org.broadinstitute.consent.http.models.User;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.Update;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class DarCollectionServiceDAO {
+
+  private final Jdbi jdbi;
+  private final UserDAO userDAO;
+
+  @Inject
+  public DarCollectionServiceDAO(Jdbi jdbi, UserDAO userDAO) {
+    this.jdbi = jdbi;
+    this.userDAO = userDAO;
+  }
+
+  public void createElectionsForDarCollection(DarCollection collection) throws SQLException {
+    final Date now = new Date();
+    jdbi.useHandle(
+        handle -> {
+          // By default, new connections are set to auto-commit which breaks our rollback strategy.
+          // Turn that off for this connection. This will not affect existing or new connections and
+          // only applies to the current one in this handle.
+          handle.getConnection().setAutoCommit(false);
```

Comment:
This would allow the entire transaction block like you mentioned in the changes notes, right?

Reply:
In and of itself, no, it just makes it so each sql is not committed after execution. The rollback operation would happen if any of the individual sql statements fail which would happen on the `commit()` line.